### ISSUE TITLE
Bulk Synchronize fails with unexpected message

### ISSUE THREAD

**Issue by ordinarytype** [zzzprojects/EntityFramework-Extensions#369]
Hello @JonathanMagnan,

We have faced an issue with bulk synchronize. Bulk operation fails with error message: 

> Table 'C:\WINDOWS\SERVIC~1\NETWOR~1\AppData\Local\Temp\#sql13ac_27_2b7' is marked as crashed and should be repaired

We're using 10.3.11-MariaDB.
Sample project is attached.

Thanks!
[BulkIssue.zip](https://github.com/zzzprojects/EntityFramework-Extensions/files/5423342/BulkIssue.zip)

**Comment by JonathanMagnan** [comment]
Hello @ordinarytype ,

We see  indeed to have an issue when appending many time very long string. We currently believe this is some limitations of using a derived table when performing the `Update`

However, I believe your importation have an issue as well. For some product id such as `64317`, the same bar code appears thousands of times.

You can easily fix the issue by doing a distinct:

```csharp
items.ForEach(x => x.Barcodes = x.Barcodes.Distinct().ToList());
```

our library should work after.

Let me know if that worked once you fixed this part in your importation. Meanwhile, we will look how to fix it on our side to make it works.

Best Regards,

Jon

**Comment by ordinarytype** [comment]
@JonathanMagnan 
Yes, after cleaning barcodes from repeatable entries & not utf-8 chars, it's started to work. 
Thanks.

**Comment by JonathanMagnan** [comment]
Awesome @ordinarytype !

We are glad everything is working out for you!

Don't hesitate to contact us with any questions, issues or feedback!

Best regards,

Jon
